CHANGELOG.md

```diff
@@ -7,6 +7,260 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [unreleased]
 
+## [15.0.0]
+
+### Added
+
+-   Added Multitenancy Recipe & always initialized by default.
+-   Adds Multitenancy support to all the recipes
+-   Added new Social login providers - LinkedIn
+-   Added new Multi-tenant SSO providers - Okta, Active Directory, Boxy SAML
+-   All APIs handled by Supertokens middleware can have an optional `tenantId` prefixed in the path. e.g. <basePath>/<tenantId>/signinup
+
+### Breaking changes
+
+-   `getUsersOldestFirst` & `getUsersNewestFirst` has mandatory parameter `tenantId`. Pass `'public'` if not using multitenancy.
+-   Added mandatory field `tenantId` to `EmailDeliveryInterface` and `SmsDeliveryInterface`
+-   Added mandatory parameter/field `tenantId` to API interfaces, Recipe interfaces and Recipe functions. Pass `'public'` if not using multitenancy.
+-   Removed deprecated config `createAndSendCustomEmail` and `createAndSendCustomTextMessage`.
+-   Added `tenantId` to `fetchValue` function in `PrimitiveClaim`, `PrimitiveArrayClaim`.
+-   TypeProvider interface is re-written
+-   In the thirdparty, thirdpartyemailpassword and thirdpartypasswordless, the providers array accepts `[]ProviderInput` instead of `[]TypeProvider`
+-   Updated `authorisationUrlGET` API
+    -   Changed: Doesn't accept `clientId` anymore and accepts `clientType` instead to determine the matching config
+    -   Added: optional `pkceCodeVerifier` in the response, to support PKCE
+-   Updated `signInUpPOST` API
+    -   Removed: `clientId`, `redirectURI`, `authCodeResponse` and `code` from the input
+    -   Instead,
+        -   accepts `clientType` to determine the matching config
+        -   One of redirectURIInfo (for code flow) or oAuthTokens (for token flow) is required
+-   Updated `appleRedirectHandlerPOST`
+    -   to accept all the form fields instead of just the code
+    -   to use redirect URI encoded in the `state` parameter instead of using the websiteDomain config.
+    -   to use HTTP 303 instead of javascript based redirection.
+-   Updated `signInUp` recipe interface function in thirdparty with new parameters:
+    -   `oAuthTokens` - contains all the tokens (access_token, id_token, etc.) as returned by the provider
+    -   `rawUserInfoFromProvider` - contains all the user profile info as returned by the provider
+-   Added `manuallyCreateOrUpdateUser` recipe function in thirdparty recipe instead, to be used in place of `signInUp`
+
```

Comment:
list all functions that have changed in sigature - go recipe wise.

Reply:
Done